pkg/utils/etcdutil/testutil.go

```diff
@@ -50,3 +53,93 @@ func NewTestSingleConfig(t *testing.T) *embed.Config {
 func genRandName() string {
 	return "test_etcd_" + strconv.FormatInt(time.Now().UnixNano()%10000, 10)
 }
+
+// NewTestEtcdCluster is used to create a etcd cluster for the unit test purpose.
+func NewTestEtcdCluster(t *testing.T, count int) (servers []*embed.Etcd, etcdClient *clientv3.Client, clean func()) {
+	re := require.New(t)
+	servers = make([]*embed.Etcd, 0, count)
+
+	cfg := newTestSingleConfig(t)
+	etcd, err := embed.StartEtcd(cfg)
+	re.NoError(err)
+	etcdClient, err = CreateEtcdClient(nil, cfg.LCUrls)
+	re.NoError(err)
+	<-etcd.Server.ReadyNotify()
+	servers = append(servers, etcd)
+
+	for i := 1; i < count; i++ {
+		// Check the client can get the new member.
+		listResp, err := ListEtcdMembers(etcdClient)
+		re.NoError(err)
+		re.Len(listResp.Members, i)
+		// Add a new member.
+		etcd2 := MustAddEtcdMember(t, cfg, etcdClient)
+		cfg2 := etcd2.Config()
+		cfg = &cfg2
+		<-etcd2.Server.ReadyNotify()
+		servers = append(servers, etcd2)
+	}
+
+	checkMembers(re, etcdClient, servers)
+
+	clean = func() {
+		etcdClient.Close()
+		for _, server := range servers {
+			if server != nil {
+				server.Close()
+			}
+		}
+	}
+
+	return
+}
+
+// MustAddEtcdMember is used to add a new etcd member to the cluster.
+func MustAddEtcdMember(t *testing.T, cfg1 *embed.Config, client *clientv3.Client) *embed.Etcd {
+	return addEtcdMemberWithRetry(t, cfg1, client, 3)
+}
+
+func addEtcdMemberWithRetry(t *testing.T, cfg1 *embed.Config, client *clientv3.Client, retry int) *embed.Etcd {
+	re := require.New(t)
+	cfg2 := newTestSingleConfig(t)
+	cfg2.Name = genRandName()
+	cfg2.InitialCluster = cfg1.InitialCluster + fmt.Sprintf(",%s=%s", cfg2.Name, &cfg2.LPUrls[0])
+	cfg2.ClusterState = embed.ClusterStateFlagExisting
+	peerURL := cfg2.LPUrls[0].String()
+	addResp, err := AddEtcdMember(client, []string{peerURL})
+	re.NoError(err)
+	// Check the client can get the new member.
+	members, err := ListEtcdMembers(client)
+	re.NoError(err)
+	re.Len(addResp.Members, len(members.Members))
+	// Start the new etcd member.
+	etcd2, err := embed.StartEtcd(cfg2)
+	if err != nil {
+		re.Contains(err.Error(), "error validating peerURLs")
```

Comment:
When will this error happen?

Reply:
In #6890, it occurs with "member count is unequal"

Reply:
Is the error expected?

Reply:
No, we have sent `MemberAdd` before starting a new etcd server, but it still returns `error validating peerURLs` with `member count is unequal`.